exchanges/orderbook/incremental_updates.go

```diff
@@ -0,0 +1,234 @@
+package orderbook
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+)
+
+// Action defines a set of differing states required to implement an incoming orderbook update used in conjunction with
+// UpdateEntriesByID
+type Action uint8
+
+const (
+	// Amend applies amount adjustment by ID
+	Amend Action = iota + 1
+	// Delete removes price level from book by ID
+	Delete
+	// Insert adds price level to book
+	Insert
+	// UpdateInsert on conflict applies amount adjustment or appends new amount to book
+	UpdateInsert
+)
+
+// Public error vars
+var (
+	ErrDepthNotFound = errors.New("orderbook depth not found")
+	ErrEmptyUpdate   = errors.New("update contains no bids or asks")
+)
+
+var (
+	errInvalidAction          = errors.New("invalid action")
+	errAmendFailure           = errors.New("amend update failure")
+	errDeleteFailure          = errors.New("delete update failure")
+	errInsertFailure          = errors.New("insert update failure")
+	errUpdateInsertFailure    = errors.New("update/insert update failure")
```

Comment:
🐒 Possibly better naming `Failure` 👉 `Failed`
🦍 Not sure if we need to differentiate anything except insert/update vs Delete. I'm easy on this.
```suggestion
	errUpdateFailed           = errors.New("orderbook update failed")
	errDeleteFailed          = errors.New("orderbook update delete failed")
```